cmd/datamon/cmd/cli_workshop_test.go

```diff
@@ -0,0 +1,494 @@
+	require.NoError(t, cmd.Process.Kill())
+	err = cmd.Wait()
+	killed = true
+	require.Equal(t, "signal: killed", err.Error(), "cmd exit with killed error")
+
+	// downloading the bundle
+	pathToDownload, _ := ioutil.TempDir("", "mmd-")
+	defer os.RemoveAll(pathToDownload)
+
+	cmd, _ = testCommand(t, false, false, targetBinary,
+		"bundle",
+		"download",
+		"--repo", repo,
+		"--label", label,
+		"--destination", pathToDownload,
+	)
+	output, err = cmd.CombinedOutput()
+	require.NoErrorf(t, err, "unexpected error, got: %s", string(output))
+
+	err = filepath.Walk(pathToDownload, func(pth string, info os.FileInfo, _ error) error {
+		rel, _ := filepath.Rel(pathToDownload, pth)
+		if info.IsDir() || model.IsGeneratedFile(rel) {
+			return nil
+		}
+		name := filepath.Base(pth)
+		require.Contains(t, files, name)
+		files[name].downloaded = true
+		assert.Equal(t, files[name].size, int(info.Size()))
+		b, erf := ioutil.ReadFile(pth)
+		require.NoError(t, erf)
+		assert.EqualValues(t, files[name].data, b)
+		return nil
+	})
+	require.NoError(t, err)
+	for name, file := range files {
+		assert.Truef(t, file.downloaded, "expected file %s to be downloaded", name)
+	}
+
+	// downloading some files
+	pathToSomeDownload, _ := ioutil.TempDir("", "mmd-")
+	defer os.RemoveAll(pathToSomeDownload)
+
+	var picked string
+	for n := range files {
+		picked = n
+		break
+	}
+
+	cmd, _ = testCommand(t, false, false, targetBinary,
+		"bundle",
+		"download",
+		"--repo", repo,
+		"--label", label,
+		"--destination", pathToSomeDownload,
+		"--name-filter", picked[:8]+".+",
+	)
+	output, err = cmd.CombinedOutput()
+	require.NoErrorf(t, err, "unexpected error, got: %s", string(output))
+
+	err = filepath.Walk(pathToSomeDownload, func(pth string, info os.FileInfo, _ error) error {
+		rel, _ := filepath.Rel(pathToSomeDownload, pth)
+		if info.IsDir() || model.IsGeneratedFile(rel) {
+			return nil
+		}
+		name := filepath.Base(pth)
+		require.Equal(t, picked, name)
+		assert.Equal(t, files[name].size, int(info.Size()))
+		b, erf := ioutil.ReadFile(pth)
+		require.NoError(t, erf)
+		assert.EqualValues(t, files[name].data, b)
+		return nil
+	})
+	require.NoError(t, err)
+
+	// no test here for web (see pkg/web)
+	// no test here for upgrade (see self_upgrade_test.go)
+}
+
+func makeTestWorkshopBundle(t testing.TB, pathToData string) map[string]*testFile {
+	files := make(map[string]*testFile, 10)
+	for i := 0; i < 9; i++ {
+		name := internal.RandStringBytesMaskImprSrc(10)
+		data := internal.RandBytesMaskImprSrc(100)
+		err := ioutil.WriteFile(filepath.Join(pathToData, name), data, 0644)
+		require.NoError(t, err)
+		files[name] = &testFile{data: data, size: 100}
+	}
+	name := internal.RandStringBytesMaskImprSrc(10)
+	data := []byte{} // empty file
+	err := ioutil.WriteFile(filepath.Join(pathToData, name), data, 0644)
+	require.NoError(t, err)
+	files[name] = &testFile{data: data, size: 0}
+	return files
+}
+
+func TestWorkshopBrewInstall(t *testing.T) {
```

Comment:
this test seems kind of loopy:

let's say the test fails because the release mechanism isn't working.  now, to fix the release mechanism, we need to make a change to the source.  we can't make the change to the source while ci is failing.

so if this test fails, i guess the fix always involve at least two merges:
* one to skip this test and get the release mechanism fixed
* one to re-enable the test

Reply:
Well this is mitigated in part because it attempts to install the _previous_ release, the one that is already in place and supposed to be working. If the current  dev breaks the next release we won't be able to see it with this test.

Indeed, if we get to break things this will start failing (similar issue with self_update_test.go btw) we need to disable the test then to reenable the test.

This is the half empty bottle.

The half full bottle indicates that we have automated CI test that validate our releases can be readily installed by users...